app/charts/shared/chart-helpers.tsx

```diff
@@ -411,3 +416,23 @@ export const useImputationNeeded = ({
 
   return imputationNeeded;
 };
+
+/** Use to potentially extract temporal values from data. This is needed for
+ * column charts, where the temporal dimension is used as X axis (and we
+ * do not fetch all values for temporal dimensions, only the min and max).
+ */
+export const useMaybeTemporalDimensionValues = (
+  dimension: DimensionMetadataFragment,
+  data: Observation[]
+) => {
+  const maybeTemporalDimensionValues: DimensionValue[] = useMemo(() => {
+    if (isTemporalDimension(dimension)) {
+      const dates = data.map((d) => d[dimension.iri] as string);
+      return uniq(dates).map((d) => ({ label: d, value: d }));
+    } else {
+      return dimension.values;
+    }
+  }, [dimension, data]);
+
+  return maybeTemporalDimensionValues;
+};
```

Comment:
I think it would be better to separate the getMaybeTemporalDimensionValues from the useMemo. Then it is easier to test (no need for renderHook, only a normal function test) and concerns (react memoizing and business logic) are not mixed. From the calling point, it is also simpler to grok what's happening.